### ISSUE TITLE
In Browserify: "Cannot find module './aes'"

### ISSUE THREAD

**Issue by ptarjan** [digitalbazaar/forge#294]
When packaging your library into a browserify project, I get 

```
"Cannot find module './aes'"
```

For a module to work in browserify, its `require()`s need to be statically analyzable. I'm not sure the easiest way to do that for you, but just putting

```
require('./aes');
require('...')
```

would probably do the trick. I know it is duplicative, so maybe you can find a way that isn't duplicating code but yet is still statically analyzable.

One way to test would probably be this:

```
$ browserify ./js/forge.js | node
module.js:338
    throw err;
          ^
Error: Cannot find module './aes'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at s ([stdin]:1:114)
    at [stdin]:1:305
    at [stdin]:32:14
    at Array.map (native)
    at Object.module.exports ([stdin]:31:21)
    at defineFunc ([stdin]:48:10)
```

And make it so that it doesn't throw anymore.

![image](https://cloud.githubusercontent.com/assets/40143/9695237/8a67371a-5314-11e5-951c-3bdb42ffb48b.png)

**Comment by dlongley** [comment]
Revamping the module system is on the [roadmap](#203) -- we jotted down some preliminary ideas here: #126.

In general, we need to get a tool chain setup that can wrap forge modules in a variety of different ways for different loaders (AMD/CommonJS/browserify/etc). Just haven't had the time to get to it yet; PRs welcome!

But, we could potentially look into the kind of change you're talking about as a near term fix for browserify should some time shake loose. We need to ensure that any change doesn't cause havoc with the other ways people are loading the library, which means better module loader tests are needed as well.

**Comment by fujifish** [comment]
for what it's worth, I was able to workaround this by generating and using the standalone forge bundle and then replacing every occurrence of `require('node-forge')` with `window.forge`.

In package.json:
```javascript
"dependencies": {
  "node-forge": "0.6.40"
},
"main": "main.js",
"scripts": {
  "forge": "(cd node_modules/node-forge && npm run bundle && cp js/forge.bundle.js ../../)",
  "build": "npm run forge && browserify main.js | sed \"s/require[(]'node-forge'[)]/window.forge/g\" > bundle.js"
}
```

Then in the HTML:
```html
<head>
  <script src="forge.bundle.js"></script>
  <script src="bundle.js"></script>
</head>
```

**Comment by dlongley** [comment]
Should hopefully be solved by https://github.com/digitalbazaar/forge/pull/357#issuecomment-247722405.

**Comment by dlongley** [comment]
Should be addressed in a new release now that #456 has been merged.
